### DISCUSSION TITLE
Mypy can't detected provider modules.

### DISCUSSION THREAD

**Discussion by zhzhang** [apache/airflow#15519] [Q&A] [answered]
When I try to type check a project with mypy, I get errors like.
```
Cannot find implementation or library stub for module named 'airflow.providers.amazon.aws.operators.sagemaker_model'
```
This is after installing the relevant provider package.

The reason this happens is because there's a missing `__init__.py` in the `airflow/providers` directory. Is there a different way to get mypy (or other type checkers) to find the module? If not, is there any issue with adding `airflow/providers/__init__.py`?

**Comment by potiuk** [comment, marked as answer]
You can run mypy with `--namespace-packages` switch as we do in Airflow CI: https://github.com/apache/airflow/blob/master/scripts/in_container/run_mypy.sh#L39
